orc8r/cloud/go/tools/swaggergen/generate/generate.go

```diff
@@ -242,6 +288,19 @@ func StripAndWriteSwaggerSpecs(specs map[string]MagmaSwaggerSpec) error {
 	return nil
 }
 
+// GetStandaloneFilepath parses the targetFilepath for its originating service.
+func GetStandaloneFilepath(targetFilePath string, rootDir string) (string, error) {
+	absTargetFilepath, err := filepath.Abs(targetFilePath)
+	if err != nil {
+		return "", errors.Wrapf(err, "target filepath %s is invalid", targetFilePath)
+	}
+	// Iterate through the file path to retrieve the service name of
+	// the Swagger spec
+	splitPath := strings.Split(absTargetFilepath, "/")
+	outFile := splitPath[len(splitPath)-4] + ".swagger.v1.yml"
```

Comment:
Still missing comment on why `len(splitPath)-4` -- when I read this in isolation I think: "why 4 ??"